### ISSUE TITLE
Linker problem

### ISSUE THREAD

**Issue by Mezek** [JohannesBuchner/cuba#5]
Gcc  version 10.2.1 during compilation throws some errors of the type:
gcc -O3 -fomit-frame-pointer -ffast-math -Wall -DHAVE_CONFIG_H -I./src/common -I. -I. -o demo-c ./demo/demo-c.c libcuba.a -lm
/usr/bin/ld: libcuba.a(Suave.o):(.bss+0x0): multiple definition of `cubafun_`; libcuba.a(Vegas.o):(.bss+0x0): first defined here
/usr/bin/ld: libcuba.a(Divonne.o):(.bss+0x0): multiple definition of `cubafun_`; libcuba.a(Vegas.o):(.bss+0x0): first defined here
/usr/bin/ld: libcuba.a(Cuhre.o):(.bss+0x0): multiple definition of `cubafun_`; libcuba.a(Vegas.o):(.bss+0x0): first defined here
/usr/bin/ld: libcuba.a(Fork.o):(.bss+0x0): multiple definition of `cubafun_`; libcuba.a(Vegas.o):(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status
make: *** [makefile:269: demo-c] Error 1

Probably new default of Gcc -fno-common is causing the problem.

**Comment by JohannesBuchner** [comment]
Could you please test if including -fcommon in https://github.com/JohannesBuchner/cuba/blob/master/makesharedlib.sh#L5 works?

**Comment by Mezek** [comment]
No, it doesn't. I've been already trying to add `extern coreinit cubafun_;` to some header files. It removes linking errors above, but then I was unable run `pycuba_demo.py`.

**Comment by JohannesBuchner** [comment]
Not sure I can be helpful here, but be sure to clear before recompiling. `make clean` should do it.

**Comment by tboschi** [comment]
I have the same issue. cubafun_ is defined in CSample.c which is included in Vegas.c, Suave.c, Divonne.c, and Cuhre.c. Creating the objects for each cuba function recompiles the CSample source.
My guess is that when creating the static library libcuba.a there is no linking and so no error. On the other hand, the shared library requires the objects to be linked, but it fails for the multiple definitions.
I don't know if you have a ready solution to this, but I am trying to fix it myself. I'll submit a pull request if I succeed

**Comment by tboschi** [comment]
I just realised that it is the veresion of cuba in this repository that is broken. I just reproduced what the makesharedlib script does (added -fPIC to the makefile) and it worked without errors.
I used this version of cuba http://www.feynarts.de/cuba/Cuba-4.2.1.tar.gz

**Comment by JohannesBuchner** [comment]
Parallelisation in later Cuba versions causes issues with Python, but Thomas Kuhn let me know that one can disable this with a Makefile flag.

@tboschi can you have a look at this and see if it works with pycuba?

**Comment by tboschi** [comment]
I am not very familiar with pycuba to be honest, but if I found some time I can have a look. Are there specific tests I can do to see if pycuba works as expected?

**Comment by jschueller** [comment]
see also https://github.com/jschueller/cuba where I added a cmake build script
